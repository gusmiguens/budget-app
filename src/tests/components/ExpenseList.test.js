const React = require('react')
const { shallow } = require('enzyme')
const { ExpenseList } = require('../../components/ExpenseList')
const expenses = require('../fixtures/expenses')

describe('<ExpenseList />', () => {
    it('Should render ExpenseList with expenses', () => {
        const wrapper = shallow(<ExpenseList expenses={expenses}/>)
        expect(wrapper).toMatchSnapshot()
    })
    it('Should render Expense list with empty message', () => {
        const wrapper = shallow(<ExpenseList expenses={[]} />)
        expect(wrapper).toMatchSnapshot()
    })
})