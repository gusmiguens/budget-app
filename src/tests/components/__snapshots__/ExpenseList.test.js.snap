// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseList /> Should render Expense list with empty message 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item-message"
    >
      <span>
        No expenses
      </span>
    </div>
  </div>
</div>
`;

exports[`<ExpenseList /> Should render ExpenseList with expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <_default
      amount={1000}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      note=""
    />
    <_default
      amount={2000}
      createdAt={-345600000}
      description="Rent"
      id="2"
      key="2"
      note=""
    />
    <_default
      amount={3000}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;
